src/annotator/util/normalize.ts

```diff
@@ -70,24 +176,55 @@ export function translateOffsets(
   start: number,
   end: number,
   filter: (ch: string) => boolean,
+  options: TranslateOffsetOptions = {},
 ): [number, number] {
-  const beforeStartCount = countChars(input, filter, 0, start);
-  const startToEndCount = countChars(input, filter, start, end);
+  start = Math.max(0, Math.min(start, input.length));
+  end = Math.max(start, Math.min(end, input.length));
+
+  const normInput = options.normalize
+    ? normalizeWithOffsets(input, { offsets: true })
+    : ({ output: input } as NormalizeResult);
+  const normOutput = options.normalize
+    ? normalizeWithOffsets(output, { reverseOffsets: true })
+    : ({ output } as NormalizeResult);
```

Comment:
Why do we need to cast to `NormalizedResult` on an inline object that does not fulfill the type contract? COuld that cause some issue?

Reply:
Perfect!